var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -187,7 +187,8 @@ def flag_handler(self, name, flags):
         cmake_flags = []
 
         if name == "cflags":
-            if "clang" in self.compiler.cc or "gcc" in self.compiler.cc:
+            cc_name = os.path.basename(self.compiler.cc)
+            if "clang" in cc_name or "gcc" in cc_name:
```

Comment:
I was going to suggest testing for `isinstance(self.compiler, spack.compilers.clang.Clang)` but apparently Aocc doesn't follow this convention. This is kind of a hack but it looks likely enough to work among the alternatives...

Reply:
I agree it is pretty hacky looking, and I'm not even sure that it is necessary to jump thru hoops like this to disable warnings. However, the way it stands is completely broken for spack-installed compilers, so I was just trying to get something in that wouldn't be reverted again.